README.md

```diff
@@ -1,49 +1,169 @@
+in your results. This project aims to provide an environment in which the stability of
+programs can be evaluated, reducing the overhead on both tool developers and consumers.
 
-## Supported Languages/Libraries
-
-- BLAS
-- LAPACK
-- Python3.6.5
-- Numpy (Python3.6.5)
-- Pytorch (Python3)
-- Libmath
 
+## Usage
 
-## Build Instructions
+#### Building & booting the environment
+You can get started with *Fuzzy* quite simply, just launch a
+[Docker](https://www.docker.com/) container as follows:
+
+```
+docker run -ti verificarlo/fuzzy
+```
+
+If you're on a shared system and would prefer to use [Singularity](https://sylabs.io/),
+that's no problem, just convert the container using the appropriate method for
+your system ([e.g.](https://docs.computecanada.ca/wiki/Singularity#Creating_images)).
+
+If you would like to build the environment locally on your system, look at the
+Dockerfiles in `docker/base/` to see how installation was performed. At the end of the
+build chain, you'll find instrumented versions of `libmath`, `lapack`, `python3`,
+`numpy`, and several other recompiled libraries.
+
+An example for how to verify your installation for Python could be the following:
+```
+$ python3 -c "print([sum([.001]*1000) for _ in range(3)])"
+[1.0, 0.9999999999999997, 1.0000000000000007, 1.0000000000000002]
+```
+
+#### Adding your software
+If your code is written in, say, Python, then you can simply mount your code to
+this environment, run it using the contained interpreter, and experince the *:fuzz:*.
+If your code loads shared libraries, such as `libmath` or `lapack`, make sure that
+they are using the instrumented versions of these libraries made available in
+`/usr/local/lib/`. You can check whether this is the case using `ldd </path/to/yourbinary>`.
+
+The next step is to make sure the environment is configured to introduce perturbations
+the way you expect. You can start with a configuration which performs perturbations
+akin to randomizing machine error with the following:
+
+```
+echo "libinterflop_mca.so -m mca --precision-binary32=24 --precision-binary64=53" > $VFC_BACKENDS_FROM_FILE
+```
+
+For more usage instructions on how to control how perturbations are introduced
+and where they occur within operations, please refer to the
+[Verificarlo](https://github.com/verificarlo/verificarlo) repository.
+
+A simple sanity check to verify that your code is using the perturbed libraries
+is to dramatically reduce the precision in your configuration by changing the
+`precision-binary32` and `precision-binary64` values to something small (e.g. 1),
+boot up a debugging session in your environment (e.g. a Python shell, GDB, etc.),
+load a math library, and run a basic math function a few times (e.g. `sin(1)`) —
+you should get different results if your instrumentation is setup properly.
+***Important***: Don't forget to set the precision back to your desired level
+prior to performing your experiments!
+
```

Comment:
How to make sure the config actually changes the precision after we generate the config? A concrete step-by-step example would be appreciated

Reply:
Thanks!